libraries/AP_RangeFinder/AP_RangeFinder_Ainstein_LR_D1.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AP_RangeFinder_config.h"
+
+#if AP_RANGEFINDER_AINSTEIN_LR_D1_ENABLED
+
+#include "AP_RangeFinder_Ainstein_LR_D1.h"
+#include <GCS_MAVLink/GCS.h>
+
+// get_reading - read a value from the sensor
+bool AP_RangeFinder_Ainstein_LR_D1::get_reading(float &reading_m)
+{
+    if (uart == nullptr) {
+        return false;
+    }
+
+    bool has_data = false;
+
+    uint32_t available = MAX(uart->available(), PACKET_SIZE*4);
+    while (available >= PACKET_SIZE) {
+        // ---------------
+        // Sync up with the header
+        const uint8_t header[] = {
+            0xEB,   // Header MSB
+            0x90,   // Header LSB
+            0x00   // Device ID
+        };
+        for (uint8_t i = 0; i<ARRAY_SIZE(header); i++) {
+            available--;
+            if (uart->read() != header[i]) {
+                continue;
+            }
+        }
+
+        const uint8_t rest_of_packet_size = (PACKET_SIZE - ARRAY_SIZE(header));
+        if (available < rest_of_packet_size) {  
+            return false;
+        }
+
+        // ---------------
+        // header is aligned!
+        // ---------------
+
+        uint8_t buffer[rest_of_packet_size];
+        available -= uart->read(buffer, ARRAY_SIZE(buffer));
+
+        const uint8_t checksum = buffer[ARRAY_SIZE(buffer)-1]; // last byte is a checksum
+        if (crc_sum_of_bytes(buffer, ARRAY_SIZE(buffer)-1) != checksum) {
+            // bad Checksum
+            continue;
+        }
+
+        const uint8_t malfunction_alert = buffer[1];
+        reading_m = UINT16_VALUE(buffer[3], buffer[4]) * 0.01;
+
+        const uint8_t snr = buffer[5];
+
+        has_data = true;
+
+        if (malfunction_alert_prev != malfunction_alert) {
+            malfunction_alert_prev = malfunction_alert;
+            report_malfunction(malfunction_alert);
+        }
+
+        /* From datasheet:
+            Altitude measurements associated with a SNR value 
+            of 13dB or lower are considered erroneous
+        */
+        signal_quality_pct = (snr <= 13 || malfunction_alert != 0) ? RangeFinder::SIGNAL_QUALITY_MIN : RangeFinder::SIGNAL_QUALITY_MAX;
+
+        if (snr == 0) {
```

Comment:
```suggestion
        if (snr == 0 || signal_quality_pct != RangeFinder::SIGNAL_QUALITY_MAX) {
```

You also need to set Status::NoData/OutOfRangeHigh/Low/Good around here somewhere